src/app/preferences/general-preferences/general-preferences.service.ts

```diff
@@ -0,0 +1,17 @@
+namespace Core {
+    export class GeneralPreferencesService {
+      constructor(private $window: ng.IWindowService) {
+        'ngInject';
+        this.$window.localStorage.setItem('current_state', "ON");
```

Comment:
As I said a local storage key is global. Use a more specific and meaningful name for it.
`current_state` -> `defaultVerticalNavState`

Reply:
For the values, maybe instead of `ON` / `OFF`, `show` / `hide` should be more clear about meaning.